src/test/java/org/example/jvspringbootfirstbook/controller/BookControllerTest.java

```diff
@@ -0,0 +1,238 @@
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class BookControllerTest {
+    public static final String TITLE = "Title";
+    public static final String AUTHOR = "author";
+    public static final String ISBN = "1231231";
+    public static final BigDecimal PRICE = BigDecimal.valueOf(123.12);
+    public static final String DESCRIPTION = "ok book";
+    public static final String COVER_IMAGE = "image";
+    public static final Set<Long> CATEGORIES_ID = Set.of();
+    public static final Set<Category> CATEGORIES = Set.of();
+    protected static MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    private BookService bookService;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired WebApplicationContext applicationContext
+    ) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @AfterAll
+    static void afterAll(
+            @Autowired DataSource dataSource
+    ) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    private static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("db/books/controller/delete-book-from-books.sql")
+            );
+        }
+    }
+
+    @WithMockUser(username = "user", roles = {"ADMIN"})
+    @Test
+    public void createBook_ValidCreateBookRequestDto_Success() throws Exception {
+        //Given
+        CreateBookRequestDto createBookRequestDto = getCreateBookRequestDto();
+        Book book = getBook();
+        BookDtoWithoutCategoryIds expected = getBookDtoWithoutCategoryIdsFromBook(book);
+
+        String jsonRequest = objectMapper.writeValueAsString(createBookRequestDto);
+        //When
+        MvcResult result = mockMvc.perform(
+                        post("/books")
+                                .content(jsonRequest)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+        //Then
+        BookDtoWithoutCategoryIds actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                BookDtoWithoutCategoryIds.class
+        );
+        Assertions.assertNotNull(actual);
```

Comment:
You can use 
```
Assertions.assertAll(
    () -> Assertions.assertNotNull(actual),...
```
All the assertions inside assertAll are executed, and if any of them fail, the test will report all failures, not just the first one encountered. This approach improves test coverage by allowing multiple failures to be detected at once.